lib/rolodex/router.ex

```diff
@@ -0,0 +1,205 @@
+
+      unquote(block)
+
+      def __router__(:phoenix_router), do: unquote(phoenix_router)
+      def __router__(:routes), do: @routes
+    end
+  end
+
+  @doc """
+  Defines a set of routes with different HTTP action verbs at the same path to document
+
+      routes [:get, :put, :delete], "/api/entity/:id"
+  """
+  defmacro routes(verbs, path) when is_list(verbs) do
+    quote do
+      unquote(Enum.map(verbs, &set_route(&1, path)))
+    end
+  end
+
+  @doc """
+  Defines a route with the given HTTP action verb and path to document
+
+      route :get, "/api/entity/:id"
+  """
+  defmacro route(verb, path) when is_atom(verb), do: set_route(verb, path)
+
+  @doc """
+  Defines an HTTP GET route at the given path to document
+
+      get "/api/entity/:id"
+  """
+  defmacro get(path), do: set_route(:get, path)
+
+  @doc """
+  Defines an HTTP POST route at the given path to document
+
+      post "/api/entity"
+  """
+  defmacro post(path), do: set_route(:post, path)
+
+  @doc """
+  Defines an HTTP PUT route at the given path to document
+
+      put "/api/entity/:id"
+  """
+  defmacro put(path), do: set_route(:put, path)
+
+  @doc """
+  Defines an HTTP PATCH route at the given path to document
+
+      patch "/api/entity/:id"
+  """
+  defmacro patch(path), do: set_route(:patch, path)
+
+  @doc """
+  Defines an HTTP DELETE route at the given path to document
+
+      delete "/api/entity/:id"
+  """
+  defmacro delete(path), do: set_route(:delete, path)
+
+  @doc """
+  Defines an HTTP HEAD route at the given path to document
+
+      head "/api/entity/:id"
+  """
+  defmacro head(path), do: set_route(:head, path)
+
+  @doc """
+  Defines an HTTP OPTIONS route at the given path to document
+
+      options "/api/entity/:id"
+  """
+  defmacro options(path), do: set_route(:options, path)
+
+  defp set_route(verb, path) do
+    quote do
+      @routes {unquote(verb), unquote(path)}
+    end
+  end
+
+  @doc """
+  Collects all the routes defined in a `Rolodex.Router` into a list of fully
+  serialized `Rolodex.Route` structs.
+  """
+  @spec build_routes(module(), Rolodex.Config.t()) :: [Rolodex.Route.t()]
+  def build_routes(router_mod, config) do
+    phoenix_router = router_mod.__router__(:phoenix_router)
+
+    router_mod.__router__(:routes)
+    |> Enum.reduce([], fn {verb, path}, routes ->
+      # If Rolodex can't find a matching Phoenix route OR if the associated
+      # controller action has no doc annotation, `build_route/4` will return `nil`
+      # and we strip it from the results
+      case build_route(verb, path, phoenix_router, config) do
+        nil -> routes
+        route -> [route | routes]
+      end
+    end)
+  end
+
+  defp build_route(verb, path, phoenix_router, config) do
+    phoenix_router
+    |> build_route_info(verb, path)
+    |> with_doc_annotation()
+    |> Rolodex.Route.new(config)
+  end
+
+  # Backwards compatibility logic for fetching Phoenix route info:
```

Comment:
What do you think about making this a breaking change, and only supporting >= 1.4.7?

Reply:
I added the backwards compatibility so we could start using this update internally right away on apps running on v1.4.6. If we get to >= v1.4.7 before this refactor ships I'm open to making it a breaking change.